tests/main/cgroup-pids/task.yaml

```diff
@@ -0,0 +1,57 @@
+summary: Each snap app and hook is tracked in a pids cgroup
+
+details: |
+    This test creates a snap process that suspends itself and ensures that it
+    placed into the appropriate hierarchy under the pids cgroup.
+
+prepare: |
+    # This feature depends on the release-app-awareness feature
+    snap set core experimental.refresh-app-awareness=true
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
+
+restore: |
+    rmdir /sys/fs/cgroup/pids/snap.test-snapd-sh.test-snapd-sh || true
+
+    # TODO: There is currently no way to unset configuration options.
+    # Once this is fixed please uncomment this line:
+    # snap unset core experimental.refresh-app-awareness
+
+execute: |
+    trap 'killall test-snapd-sh || true' EXIT
+    # Start a "sleep" process in the background
+    #shellcheck disable=SC2016
+    test-snapd-sh -c 'touch $SNAP_DATA/1.stamp && exec sleep 1h' &
+    pid1=$!
+    # Ensure that snap-confine has finished its task and that the snap process
+    # is active. Note that we don't want to wait forever either.
+    for _ in $(seq 30); do
+        test -e /var/snap/test-snapd-sh/current/1.stamp && break
+        sleep 0.1
+    done
+    # While the process is alive its PID can be seen in the cgroup.procs file
+    # of the pid controller.
+    MATCH "$pid1" < /sys/fs/cgroup/pids/snap.test-snapd-sh.test-snapd-sh/cgroup.procs
+
+    # Start a second process so that we can check adding tasks to an existing
+    # control group.
+    #shellcheck disable=SC2016
+    test-snapd-sh -c 'touch $SNAP_DATA/2.stamp && exec sleep 1h' &
+    pid2=$!
+    for _ in $(seq 30); do
+        test -e /var/snap/test-snapd-sh/current/2.stamp && break
+        sleep 0.1
```

Comment:
and here

Reply:
This is adapted from `cgroup-freezer`, its works there.